### ISSUE TITLE
Fix false negatives for except: ["after-closing-brace"] in block-closing-brace-empty-line-before

### ISSUE THREAD

**Issue by Jamiewarb** [stylelint/stylelint#4129]
> Clearly describe the bug

The `"block-closing-brace-empty-line-before"` rule with the `"except": "after-closing-brace"` does not seem to be working any longer. Rather, the behaviour seems to be identical, whether or not the except is added.

> Which rule, if any, is the bug related to?

It is specifically relating to the single exception, `"after-closing-brace"` that can be added to `"block-closing-brace-empty-line-before"`.

> What CSS is needed to reproduce the bug?

e.g.

```scss
.a {

    .b {
        cursor: pointer;
    }

} // Unexpected empty line before closing brace

.a {

    &__b {
        cursor: pointer;
    }

} // Unexpected empty line before closing brace

.a {

    &__b {
        cursor: pointer;
    }
} // No warnings shown here
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "block-closing-brace-empty-line-before": ["never", {
      "except": ["after-closing-brace"]
    }]
  }
}
```

> Which version of stylelint are you using?

9.3.0 and 10.1.0 - Occurring on both.

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Via CLI: 

```bash
stylelint 'resources/**/*.(s?(a|c)ss|vue)' --syntax scss
```

and via Webpack: 
```javascript
plugins: [
    new StyleLintPlugin({
        configFile: '.stylelintrc',
        context: 'resources/assets',
        files: '**/*.(s?(a|c)ss|vue)',
        syntax: 'scss',
    }),
]
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS nested properties.

> What did you expect to happen?

One warning should have flagged for the third example above. No warnings should be flagged for the first two examples, as they correctly have a new line after the closing brace, as per the docs.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings are flagged:

```shell
test.scss
✖  7:1 Unexpected empty line before   block-closing-brace-empty-line-before
           closing brace
✖  15:1 Unexpected empty line before   block-closing-brace-empty-line-before
           closing brace
```

**Comment by Jamiewarb** [comment]
It seems this was also reported 13 days ago, but closed as the issue didn't follow the template:
#4112 

I believe the original addition of the "except" for this rule was discussed here #2090

**Comment by evilebottnawi** [comment]
StyleLintPlugin uses very old version of stylelint, can you reproduce problem using CLI?

**Comment by Jamiewarb** [comment]
Yes I am able to reproduce in CLI with both versions mentioned above:
`Via CLI: yarn stylelint 'resources/**/*.(s?(a|c)ss|vue)' --syntax scss`

**Comment by Jamiewarb** [comment]
Hey @evilebottnawi 
Just a nudge to see if you're able to get round to this at all?

**Comment by hudochenkov** [comment]
It never worked for rule inside another rule. It was always working for a rule inside at-rule.

https://github.com/stylelint/stylelint/blob/d1f07846ff6a1fce028b8db0b47f9aa8fa3c058d/lib/rules/block-closing-brace-empty-line-before/index.js#L72

Because CSS doesn't have rules inside another rules yet.

**Comment by Jamiewarb** [comment]
Ahh yes that makes a lot of sense, thanks for the explanation. So I guess this is something I should bring up with the SCSS plugin for stylelint?

**Comment by RyanDeegan** [comment]
@Jamiewarb did you ever get a solution to this problem? Having the same issue just now.

**Comment by Jamiewarb** [comment]
Hey @RyanDeegan unfortunately not

**Comment by RyanDeegan** [comment]
> Hey @RyanDeegan unfortunately not

No worries, thanks for getting back to me.
